WordPress/gutenberg
# Test loader for PHPUnit9 compatibility

## What?
Adds a `GutenbergTestSuiteLoader` loader to support WordPress naming conventions in PHPUnit 9.

## Why?
Calling `composer install` in the gutenberg directory now installs PHPUnit 9 which changed the rules of loading test classes.

PHPUnit 8 can correctly load a `WP_HTML_Tag_Processor_Test` class from a file called `wp-html-tag-processor-test.php`.

PHPUnit 9 expects the class to be named after the file, so `wp-html-tag-processor-test`. Alternatively, the file could be renamed to `WP_HTML_Tag_Processor_Test.php`. Neither is aligned with WordPress naming conventions. 

## How?

This PR adds a loader that helps PHPUnit 9 load Gutenberg test classes correctly.

**Before**
```
$  ./vendor/bin/phpunit  --no-configuration ./phpunit/html/wp-html-tag-processor-standalone-test.php
 
Class 'wp-html-tag-processor-standalone-test' could not be found in 'gutenberg/phpunit/html/wp-html-tag-processor-standalone-test.php'.
```

**After**
```
$ ./vendor/bin/phpunit --loader phpunit\\GutenbergTestSuiteLoader --no-configuration ./phpunit/html/wp-html-tag-processor-standalone-test.php        

PHPUnit 9.5.25 #StandWithUkraine

Warning:       Using a custom test suite loader is deprecated

...............................................................  63 / 108 ( 58%)
.............................................                   108 / 108 (100%)

Time: 00:00.021, Memory: 6.00 MB

OK (108 tests, 150 assertions)

```


## Testing Instructions

Run this locally and confirm it runs the tests:

```
./vendor/bin/phpunit --loader phpunit\\GutenbergTestSuiteLoader --no-configuration ./phpunit/html/wp-html-tag-processor-standalone-test.php        
```

cc @dmsnell @georgeh @jorgefilipecosta
